src/app/public/modules/dock/dock-item.ts

```diff
@@ -0,0 +1,37 @@
+import {
+  Observable,
+  Subject
+} from 'rxjs';
+
+/**
+ * Represents a single item added to the dock.
+ */
+export class SkyDockItem<T> {
+
+  /**
+   * An event that emits when the item is removed from the dock.
+   */
+  public get destroyed(): Observable<void> {
```

Comment:
Why not use an `EventEmitter` instead of an `Observable`?

Reply:
Generally, `EventEmitter` should be used for component's `Output` properties; `Observable` is a bit more versatile otherwise. https://angular.io/api/core/EventEmitter#eventemitter